fdbserver/DDShardTracker.actor.cpp

```diff
@@ -210,6 +210,29 @@ std::pair<ShardSizeBounds, bool> calculateShardSizeBounds(
 	return { bounds, readHotShard };
 }
 
+ACTOR Future<Void> shardUsableRegions(DataDistributionTracker::SafeAccessor self, KeyRange keys) {
+	ASSERT(SERVER_KNOBS->SHARD_ENCODE_LOCATION_METADATA);
+	ASSERT(SERVER_KNOBS->DD_SHARD_USABLE_REGION_CHECK_RATE > 0);
+	wait(yieldedFuture(self()->readyToStart.getFuture()));
+	double expectedCompletionSeconds = self()->shards->size() * 1.0 / SERVER_KNOBS->DD_SHARD_USABLE_REGION_CHECK_RATE;
```

Comment:
The comment above says this knob could be set to 0, which, if set, will definitely cause divide by 0 error and the above assertion failure.

Reply:
We check the value of DD_SHARD_USABLE_REGION_CHECK_RATE should not be 0 at line 214. This actor is spawned only if DD_SHARD_USABLE_REGION_CHECK_RATE is not 0.